toolkit/tools/pkg/imagecustomizerlib/cleanUpImage.go

```diff
@@ -0,0 +1,48 @@
+package imagecustomizerlib
+
+import (
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
+	"github.com/sirupsen/logrus"
+)
+
+func cleanUpImage(imageChroot *safechroot.Chroot) error {
+
+	logger.Log.Infof("Cleaning up image")
+
+	// clear tdnf cache
+	tdnfClean := func() error {
+		tdnfArgs := []string{
+			"-v", "clean", "all",
+		}
+		stdoutCallback := func(line string) {
+			logger.Log.Trace(line)
+		}
+		return shell.NewExecBuilder("tdnf", tdnfArgs...).
+			StdoutCallback(stdoutCallback).
+			LogLevel(shell.LogDisabledLevel, logrus.DebugLevel).
+			ErrorStderrLines(1).
+			Execute()
+	}
+
+	// Clear journal logs
+	clearJournalLogs := func() error {
+		return shell.NewExecBuilder("journalctl", "--vacuum-time=1s").
```

Comment:
With MIC, cleaning up the journald logs isn’t really necessary since the OS has never run and therefore has never had a change to produce logs.

The one case where this might be interesting is when customizing an image that has already been booted. This won’t be a common scenario. But it has come up at least once before.

However, this does touch on what I mentioned previously: MIC doesn’t make changes unless the user explicitly asks for them. So, any additional cleanup we do for the user will need to be attached to a new API. As such I recommend dropping the journald stuff from this PR.

Reply:
Got it, dropped the journald stuff from this PR.